src/unix/os390-syscalls.c

```diff
@@ -299,6 +299,11 @@ int epoll_wait(uv__os390_epoll* lst, struct epoll_event* events,
 
     ev.fd = pfd->fd;
     ev.events = pfd->revents;
+    if (i == lst->size - 1)
+      ev.is_msg = 1;
+    else
+      ev.is_msg = 0;
+
```

Comment:
how about:

set 
`ev.is_msg = 0;` within the loop, and


```c
if (i == lst->size - 1)
  events[nevents - 1].is_msg = 1;
```
after the loop? that way we reduce the number of checks to 1 from `i`.

Reply:
Sure, let me take a look at that. I think it'll require a bit more changes since it's not guaranteed that the last entry in the events array belongs to the message queue. It may be skipped because of the following condition:
```c
    pfd = &pfds[i];
    if (pfd->fd == -1 || pfd->revents == 0)
      continue;
```

Reply:
@gireeshpunathil Please see my latest commit https://github.com/libuv/libuv/pull/2013/commits/417d2defc5785a51872824c4296092576b4c97f7. I moved the check out of the loop and added some additional logic to make sure we haven't skipped the message queue FD.